### ISSUE TITLE
fix escape in ps_1771_update_customer_note()

### ISSUE THREAD

**Issue by ang3lx** [PrestaShop/PrestaShop#23633]
#### Describe the bug

error on PS update when in note field there is "
“[ERROR] PHP 1.7.7.1 / * PHP: ps_1771_update_customer_note (); * /”

#### Expected behavior

`html_entity_decode($note['note'])`
fix with 
`html_entity_decode(pSQL($note['note']))`

#### Steps to Reproduce

1. install ps 1.7.7.1
2. create user with note that contains quote/double quote
3. update prestashop to latest
4. got error ps_1771_update_customer_note()

**Screenshots**
here's the full code:

```
function ps_1771_update_customer_note()
{
    $notes = Db::getInstance()->executeS(
        'SELECT id_customer, note FROM ' . _DB_PREFIX_ . 'customer
        WHERE note IS NOT NULL AND note != ""'
    );

    $result = true;
    foreach ($notes as $note) {
        $result &= Db::getInstance()->execute(
            'UPDATE ' . _DB_PREFIX_ . 'customer
            SET note = "' . html_entity_decode($note['note']) . '"
            WHERE id_customer = ' . $note['id_customer']
        );
    }

    return (bool) $result;
}
```

It should be:

```
function ps_1771_update_customer_note()
{
    $notes = Db::getInstance()->executeS(
        'SELECT id_customer, note FROM ' . _DB_PREFIX_ . 'customer
        WHERE note IS NOT NULL AND note != ""'
    );

    $result = true;
    foreach ($notes as $note) {
        $result &= Db::getInstance()->execute(
            'UPDATE ' . _DB_PREFIX_ . 'customer
            SET note = "' . html_entity_decode(pSQL($note['note'])) . '"
            WHERE id_customer = ' . $note['id_customer']
        );
    }

    return (bool) $result;
}
```

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: 7.2

**Comment by hibatallahAouadni** [comment]
Hello @ang3lx 

Could you please create a PR (Pull Request) with your fix and edit your issue **following the provided template to submit a complete bug report** (fill all the blocks: steps to reproduce, PS version, PHP version, describe the bug, ...).

Thanks!

**Comment by hibatallahAouadni** [comment]
Hello @ang3lx ,

Since we had no news from you for more than 20 days, I'll close this ticket to avoid cluttering up the backlog.
Please note that you can always create a new one if further information pops up.

Thanks!

**Comment by dali-rajab** [comment]
Yes you are right @ang3lx ! when i tryed to upgrade my shop from 1.7.6.9 to 1.7.7.3 i got this error (it's not a blocking error by the way) :
![ps_1771_update_customer_note](https://user-images.githubusercontent.com/39015823/116358167-3e5c6200-a7f5-11eb-8f36-9dad92ba5002.jpg)
 and your fix is good ✌️😃 🎆 🚀 ❤️

**Comment by dali-rajab** [comment]
@hibatallahAouadni could you reopen this Issue please ? thank you 😃

**Comment by hibatallahAouadni** [comment]
Hello @dali-rajab 

Could you please provide steps to reproduce the issue :pray: 
If I reproduce the issue I will reopen it :wink: 

Thanks!

**Comment by ang3lx** [comment]
@hibatallahAouadni in first post i've added all infos where bug is and how fix..

1) install ps 1.7.7.1
2) create user with note that contains quote/double quote
3) update prestashop to latest
4) got error ps_1771_update_customer_note()

**Comment by dali-rajab** [comment]
exactly @ang3lx ! thank you 🚀

**Comment by PierreRambaud** [comment]
I reopen, the fix is legit :)

**Comment by hibatallahAouadni** [comment]
Hello @ang3lx & @dali-rajab 

Could you please check my screen record below and see if I miss step or something cause I didn't manage to reproduce the issue:

https://drive.google.com/file/d/1K0gpCnXEi0ZG2CRGqQ17k85ynSaJMo9J/view

Please check and feedback.
Thanks!

**Comment by ang3lx** [comment]
Hello @hibatallahAouadni 

your steps are ok, but I've got this error 
In my case, into note field there were some Japanese characters. I don't know if this is relevant.

After error i've re-run ps_1771_update_customer_note() from standalone php patch file adding pSQL() to $note['note'] and everything worked perfectly

**Comment by hibatallahAouadni** [comment]
Hello @ang3lx 

Ok let's try with Japanese characters :wink: 
Could you please provide the sentence that throw the error.

Thanks!

**Comment by ang3lx** [comment]
@hibatallahAouadni 

unfortunately (mea culpa) I have not marked the id of the note on which the update gave me an error, and at the moment there are over 10 thousand notes.
I don't think I can pass you the note that made us go wrong

try with this one

（别吓！）这个客人就是一个”小“女儿，她有“一点”胖。
脖子的周围40cm，不要胸棉。我怕客人想骗我们买一件礼服当为婚纱。。。
颜色为象牙色，腰带和花做成lavender

**Comment by dali-rajab** [comment]
@hibatallahAouadni @ang3lx , in fact, as i already mentioned in https://github.com/PrestaShop/PrestaShop/pull/24243 , you have to redo the same test but with a version < 1.7.7.1 (1.7.6.9 in my case) . thanks.

**Comment by hibatallahAouadni** [comment]
Hello @ang3lx & @dali-rajab 

I didn't manage to reproduce the issue, please check my screen record below:

https://drive.google.com/file/d/1i3xQ-OG4t1qE58cVIJoGVz1JTNaCF8Hz/view

1. install 1.7.6.9
2. add the customer note
3. upgrade my shop to 1.7.7.3
4. successful upgrade

Please check and feedback.
Thanks!

**Comment by dali-rajab** [comment]
@hibatallahAouadni sorry to tell you that you need to redo the test, but with a note that containts some double quotes `"` , in fact 
 i think the issue has nothing to do with Japanese characters, only double quotes matter here. Sorry again 😞

**Comment by hibatallahAouadni** [comment]
Hello @dali-rajab @ang3lx 

Thanks for the details!
I was able to reproduce the issue with PrestaShop version 1.7.6.9 uprading to 1.7.7.4. I’ll add this to the debug backlog.
As you said it's not blocking error, the upgrade finish with success:

https://drive.google.com/file/d/1FgQ4BrNytgrxZG5_Ta7H4KbPRZQEVxCH/view

![image](https://user-images.githubusercontent.com/23308427/117446543-83db0680-af3c-11eb-97f7-f45ccc1a5651.png)

I add these notes to my shop 1.7.5.1 and upgrade to 1.7.6.9, the upgrade process complete successfully, so it's a regression:

![image](https://user-images.githubusercontent.com/23308427/117451990-85f49380-af43-11eb-9072-76d2c9063372.png)

Thanks for your PR :rocket:

**Comment by Progi1984** [comment]
Fixed by @dali-rajab in #24243

**Comment by dali-rajab** [comment]
thanks @ang3lx
